### PR TITLE
[v16] Support custom hardware key prompt

### PR DESCRIPTION
**Pull request by gzdunek**
Backport https://github.com/gravitational/teleport/pull/47273 to branch/v16

Manual backport because of conflicts in `lib/client/keystore.go`.

### CODE REVIEW

**Comment by gzdunek** [comment]
Friendly ping @nklaassen, @Joerger.

FYI I'm backporting hardware key prompts in Connect to v16 so that's why I opened this PR.

**Review by ravicious — APPROVED** [review]

File: api/utils/keys/privatekey.go [hunk 1]
```diff
@@ -252,16 +276,16 @@ func LoadKeyPair(privFile, sshPubFile string) (*PrivateKey, error) {
 		return nil, trace.ConvertSystemError(err)
 	}
 
-	priv, err := ParseKeyPair(privPEM, marshalledSSHPub)
+	priv, err := ParseKeyPair(privPEM, marshalledSSHPub, customPrompt)
```
Comment (codingllama) [inline api/utils/keys/privatekey.go:279]: Maybe rename `marshaledSSHPub` so it matches master? Super minor though.

**Review by codingllama — APPROVED** [review]
Looks reasonable. Are there any important changes, in relation to master, that we should look out for here?

**Review by russjones — APPROVED** [review]

**Comment by gzdunek** [comment]
>Are there any important changes, in relation to master, that we should look out for here?

The file with the most significant conflicts was `lib/client/keystore.go` mainly due to changes in `privatekey.go`, where `keys.LoadKeyPair()`, `keys.ParseKeyPair()`, and `keys.ParsePrivateKey()` now take one more argument. On master, these arguments need to be passed in several places in `lib/client/keystore.go`, but in v16, it's simpler with just one call.

**Comment by gzdunek** [comment]
>Are there any important changes, in relation to master, that we should look out for here?

The file with the most significant conflicts was `lib/client/keystore.go`. In this file, we call functions from `privatekey.go` (`keys.LoadKeyPair()`, `keys.ParseKeyPair()`, and `keys.ParsePrivateKey()`) which take now one more argument. On master, these arguments need to be passed in several places in `lib/client/keystore.go`, but in v16, it's simpler with just one call.
I went through that file and it looks good, so I'm going to merge this backport.

**Pull request merged**